src/couch_replicator_httpc.erl

```diff
@@ -222,16 +222,23 @@ clean_mailbox(_ReqId, 0) ->
 clean_mailbox({ibrowse_req_id, ReqId}, Count) when Count > 0 ->
     case get(?STREAM_STATUS) of
         {streaming, Worker} ->
-            ibrowse:stream_next(ReqId),
-            receive
-                {ibrowse_async_response, ReqId, _} ->
-                    clean_mailbox({ibrowse_req_id, ReqId}, Count - 1);
-                {ibrowse_async_response_end, ReqId} ->
+            case is_process_alive(Worker) of
+                true ->
+                    ibrowse:stream_next(ReqId),
```

Comment:
The extra case for is_process_alive is making things rather deep. Can you extract the body of the true clause into something like discard_message(ReqId, Worker, Count) and then just move that below clean_mailbox.